### PR TITLE
doc: posix arch and boards: refactor and add much more content

### PR DESCRIPTION
**Pull request by aescolar**
Refactor the POSIX arch documentation out of the native_posix page into its own page.
Originally native_posix was the only POSIX arch based board, but this has not been the case anymore for several years. 
The architecture part of the documentation being in the native_posix doc has created unneeded confusion over time.

&

Added much more content describing the posix boards specially the nrf52_bsim.

### CODE REVIEW

File: boards/posix/doc/bsim_boards_design.rst [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+    the embedded code, but without any specific HW. In that way, many embedded
+    components which are dependent on the HW would not be suited for testing in
+    that platform. Just like the bsim boards, this Zephyr target board can
+    be used with or without Zephyr's ztest system and twister.
+    The native_posix board shares the :ref:`POSIX architecture<Posix arch>`
+    with the bsim boards.
+
+- Zephyr's ztest infrastructure and Zephyr's twister:
+  Based on dedicated embedded test applications build with the code under test.
+  The embedded test application is responsible to drive the tests and check the
```
Comment (daor-oti) [inline boards/posix/doc/bsim_boards_design.rst:94]: 'to drive' -> 'for driving'

File: boards/posix/doc/arch_soc.rst [hunk 2]
```diff
@@ -0,0 +1,409 @@
...
+.. contents::
+   :depth: 1
+   :backlinks: entry
+   :local:
+
+Overview
+********
+
+The POSIX architecture, in combination with the inf_clock SOC layer,
+provides the foundation,  architecture and SOC layers for a set of virtual test
```
Comment (daor-oti) [inline boards/posix/doc/arch_soc.rst:15]: 'foundation,  ' have one space too much after the ,

File: boards/posix/doc/arch_soc.rst [hunk 3]
```diff
@@ -0,0 +1,409 @@
...
+Eventually one of these HW models will raise an interrupt to the simulated CPU.
+When the IRQ controller wants to wake the simulated CPU, the HW thread is
+blocked, and the simulated CPU is awaken by letting the last SW thread continue
+executing.
+
+This process of getting the CPU to sleep, letting the HW models run,
+and raising an interrupt which wake the CPU again is repeated until the end
+of the simulation, where the CPU execution always takes 0 simulated time.
+
+When a SW thread is waken by an interrupt, it will be made to enter the
```
Comment (daor-oti) [inline boards/posix/doc/arch_soc.rst:354]: 'waken' -> 'awakened'

File: boards/posix/doc/bsim_boards_design.rst [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+
+Test code may also communicate with other devices' test code using the bsim
+backchannels. These provide a direct, reliable pipe between devices' test code
+can use to exchange data.
+
+
+About using Zephyr APIs
+=======================
+
+Note that even that bsim board code is linked with the Zephyr kernel,
```
Comment (daor-oti) [inline boards/posix/doc/bsim_boards_design.rst:212]: 'Note that even that bsim' -> 'Note that even though bsim'

**Review by daor-oti — CHANGES_REQUESTED** [review]
Looks good, but just a few minor comments.

**Review by daor-oti — DISMISSED** [review]
LGTM.

File: boards/posix/doc/bsim_boards_design.rst [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+:ref:`POSIX arch based board<posix_arch_rationale>`, but in addition these
+will include some HW models which, to some degree, pretend to be the real
+embedded HW.
+
+These tests are run in workstation, that is, without using real embedded HW.
+The intention being to be able to run tests much faster than real time,
+without the need for real HW, and in a deterministic/reproducible fashion.
+
+Unlike native_posix, bsim boards do not interact directly with any host
+peripherals, , and their execution is independent of the host load, or timing.
```
Comment (wopu-ot) [inline boards/posix/doc/bsim_boards_design.rst:43]: ", , " -> ", "

**Review by wopu-ot — DISMISSED** [review]
A tiny nitpick, but LGTM

**Review by wopu-ot — APPROVED** [review]

**Review by daor-oti — APPROVED** [review]

**Comment by ElvarT** [comment]
Thanks for this. It helps understanding this area much better.

**Pull request merged**